### DISCUSSION TITLE
Can MONAI dataloader use simpleITK for loading nifti data?

### DISCUSSION THREAD

**Discussion by fengling0410** [Project-MONAI/MONAI#6335] [Q&A]
Hi there,

I'm working with our in-house organ segmentation pipeline which is based on the MONAI package. But we got some alignment issue for scans and the predicted labelmap.

The first step of our pipeline is to extract nifti file from DICOM scans, and the nifti files are saved with SimpleITK, with the following code: 

`
def sitk_2_nifti(dicom_dir,
                 output_dir=None, 
                 filename=None,
                 skip_write = True):

    """Pass temporary symlinked directory of dicom scans to simpleitk filereader and convert to nifti file.

    Parameters:
    -----------
    scan: numpy array
        Array to be saved as a NIFTI file.
    output_dir: string
        Path to save the NIFTI file.
    filename: string
        Name of the NIFTI file to be saved.
    skip_write: bool
        skip writing if the file still exist
    """
    ### Ling: Do not write the file if it is already there
    if '.nii.gz' not in filename:
        filename += '.nii.gz'
    
    if skip_write and os.path.isfile(os.path.join(output_dir, filename)):
        print("file exists, do not overwrite")
        pass
    elif os.path.isfile(os.path.join(output_dir, filename)):
        os.remove(os.path.join(output_dir, filename))
        reader = sitk.ImageSeriesReader()
        dicom_names = reader.GetGDCMSeriesFileNames(dicom_dir)
        reader.SetFileNames(dicom_names)
        image = reader.Execute()
        sitk.WriteImage(image, os.path.join(output_dir, filename))
    else:
        if not os.path.exists(output_dir):
            os.makedirs(output_dir)
        reader = sitk.ImageSeriesReader()
        dicom_names = reader.GetGDCMSeriesFileNames(dicom_dir)
        reader.SetFileNames(dicom_names)
        image = reader.Execute()
        sitk.WriteImage(image, os.path.join(output_dir, filename))
`

After saving all the nifti files with simpleITK, these files are loaded to our model using MONAI dataloader. As I noticed, the MONAI dataloader makes use of Nibabel for loading nifti files, but simpleITK (LPS) is not consistent with Nibabel (RAS) in terms of coordinate system. Here are the visualization of numpy arrays loaded from MONAI dataloader and simpleITK reader:

<img width="454" alt="Screen Shot 2023-04-11 at 2 16 56 PM" src="https://user-images.githubusercontent.com/89789702/231254708-ea01763b-9ee8-40fe-aeaa-1969edd3925c.png">

<img width="450" alt="Screen Shot 2023-04-11 at 2 16 44 PM" src="https://user-images.githubusercontent.com/89789702/231254725-68f8ea22-745d-47cd-8635-f60d7e2ce4ed.png">

It seems like the numpy array loaded from MONAI dataloader is a rotated version of that loaded from simpleITK image reader. Therefore, the predicted labelmap output (saved using simpleITK) from our model is also a rotated version of the original scan. When we visualize the scan and labelmap using 3D slicer, the alignment issue appears:

<img width="817" alt="Screen Shot 2023-04-11 at 2 27 33 PM" src="https://user-images.githubusercontent.com/89789702/231255945-0facc4a1-0a8c-4a35-8a3b-95ff576fcd1e.png">

The MONAI data transformation is used is:
`
def get_transforms():
    """
    Initialize transforms for inference pipeline.

    Returns:
    --------
    torch transforms:
        Training transform pipelines for inference

    """
    test_transforms = Compose(
        [
            LoadImaged(keys=["image"], ensure_channel_first=True),
            ScaleIntensityRanged(
                keys=["image"], a_min=-160, a_max=240, b_min=0.0, b_max=1.0, clip=True
            ),
            # Orientationd(keys=["image"], axcodes="RAS")
        ]
    )

    return test_transforms
`

I've also tried to add  Orientationd(keys=["image"], axcodes="RAS") but that doesn't really solve the problem. It just makes the labelmap rotate in another direction and still mis-align with the scan. 

May I ask for some possible reasons on why this happens and maybe also some suggestions about the next step? Thanks in advance! :)

**Comment by wyli** [comment]
please try `img = LoadImage(image_only=True, reader="ITKReader", channel_dim=None, affine_lps_to_ras=False)(filename)` 

https://github.com/Project-MONAI/MONAI/blob/fa7411a62fce8d5883edd0e4e1053008e7c6170a/monai/data/image_reader.py#L182-L188

for more examples please see https://github.com/Project-MONAI/tutorials/blob/main/modules/load_medical_images.ipynb
